[libbeat] LDAP processor improvements (#47827)
elastic/beats: docs/reference/filebeat/processor-translate-guid.md

  stack: ga
---

# Translate LDAP Attribute [processor-translate-ldap-attribute]

The `translate_ldap_attribute` processor translates LDAP attributes into friendlier values. The typical use case is converting an Active Directory Global Unique Identifier (GUID) into a human-readable name (for example the object's `cn`).

Every object on an Active Directory or an LDAP server is issued a GUID. Internal processes refer to their GUID’s rather than the object's name and these values sometimes appear in logs.

If the search attribute is invalid (malformed) or does not map to any object on the domain the processor returns an error unless `ignore_failure` is set.

The result of this operation is an array of values, given that a single attribute can hold multiple values.

> Note: the search attribute is expected to map to a single object. If multiple entries match, only the first entry's mapped attribute values are returned.

Reviewer: ```suggestion
The search attribute is expected to map to a single object. If multiple entries match, only the first entry's mapped attribute values are returned.
```

| --- | --- | --- | --- |
| `field` | yes |  | Source field containing a GUID. |
| `target_field` | no |  | Target field for the mapped attribute value. If not set it will be replaced in place. |
| `ldap_domain` | no |  | DNS domain name (e.g., `example.com`) used for DNS SRV discovery and to construct FQDNs from `LOGONSERVER`. When omitted Beats inspects OS metadata to infer the domain (Windows: `USERDNSDOMAIN`, `GetComputerNameEx`, TCP/IP + Kerberos registry keys, hostname; Linux/macOS: `/etc/resolv.conf`, `/etc/krb5.conf`, hostname). |

> | `ldap_address` | no |  | LDAP server address (e.g., `ldap://ds.example.com:389`). When omitted Beats auto-discovers controllers by querying `_ldaps._tcp.<domain>` first, `_ldap._tcp.<domain>` second, and finally the Windows `LOGONSERVER` variable if available. Candidates are tried in order until one succeeds. |

Reviewer: ```suggestion
| `ldap_address` | {applies_to}`stack: ga 9.2.3` no<br>{applies_to}`stack: ga 9.0.0` yes |  | LDAP server address (for example, `ldap://ds.example.com:389`). When omitted Beats auto-discovers controllers by querying `_ldaps._tcp.<domain>` first, `_ldap._tcp.<domain>` second, and finally the Windows `LOGONSERVER` variable if available. Candidates are tried in order until one succeeds. |
```

| `ldap_bind_user` | no |  | LDAP DN/UPN for simple bind. When provided with `ldap_bind_password` Beats performs a standard bind. When set without a password Beats issues an unauthenticated bind using this identity (useful for servers that expect a bind DN even for anonymous operations). |
| `ldap_bind_password` | no |  | LDAP password for simple bind. When both the username and password are omitted Beats attempts automatic authentication: on Windows it first tries SSPI with the Beat's service or user identity using the SPN `ldap/<hostname derived from ldap_address>` and falls back to an unauthenticated bind if that fails. Non-Windows platforms immediately use an unauthenticated bind. |
| `ldap_search_attribute` | yes | `objectGUID` | LDAP attribute to search by. |
| `ldap_mapped_attribute` | yes | `cn` | LDAP attribute to map to. |
| `ldap_search_time_limit` | no | 30 | LDAP search time limit in seconds. |

> | `ldap_ssl`* | no |  | LDAP TLS/SSL connection settings. See [SSL](/reference/filebeat/configuration-ssl.md). |

Reviewer: Is the default for this setting changing? It also looks like the contents of the footnote was moved up into the table so we can remove the `*` in the first cell.

```suggestion
| `ldap_ssl` | no | {applies_to}`stack: ga 9.2.3` no default<br>{applies_to}`stack: ga 9.0.0` `30` | LDAP TLS/SSL connection settings. Refer to [SSL](/reference/filebeat/configuration-ssl.md). |
```

| `ad_guid_translation` | no | `auto` | Controls GUID binary conversion for Active Directory attributes. `auto` (default) converts when the LDAP search attribute equals `objectGUID` (case-insensitive). Use `always` to force conversion or `never` to disable it. |
| `ignore_missing` | no | false | Ignore errors when the source field is missing. |
| `ignore_failure` | no | false | Ignore all errors produced by the processor. |

## Authentication flow

Beats attempts LDAP authentication in the following order:

> 1. Simple bind using `ldap_bind_user` + `ldap_bind_password` when both are supplied.

Reviewer: ```suggestion
1. Simple bind using `ldap_bind_user` and `ldap_bind_password` when both are supplied.
```
